Resolves #76 Decouple post_migrate signal from refreshing the views

themrinalsinha opened the pull request:
### Updated
Added `MATERIALIZED_VIEWS_DISABLE_SYNC_ON_MIGRATE` (boolean) as a settings option. When set to True, it skips running `sync_pgview` during migrations.

mikicz commented:
Hello, two things

Could you please add this setting to README?
Could you please try to write a test for this?

themrinalsinha commented:
> Hello, two things
> 
> Could you please add this setting to README? Could you please try to write a test for this?

added testcase and updated readme

sidmitra commented on tests/test_project/settings/base.py:
```diff
@@ -174,3 +174,5 @@
     },
 }
+
+MATERIALIZED_VIEWS_SYNC_DISABLED = False
```
This might need to have some common prefix for this and future environment variables.

Eg. `DJANGO_PGVIEWS_*`
`DJANGO_PGVIEWS_SYNC_POST_MIGRATE=True` 

The other options for projects is to group them into a dict. 

```
DJANGO_PGVIEWS_SETTINGS = {
    "sync_post_migrate": True
}
```

@mikicz might be able to determine what naming makes sense.

themrinalsinha replied:
Agree @sidmitra, There is an existing setting option, `MATERIALIZED_VIEWS_CHECK_SQL_CHANGED`. I aimed to keep the new setting aligned by using the same prefix for consistency.

sidmitra commented on tests/test_project/viewtest/tests.py:
```diff
@@ -438,3 +439,61 @@ def test_no_schema_list(self):
+        test execution. To address this, we store the original receivers and settings,
+        then restore them in tearDown to avoid affecting other tests.
+        """
+        from django.db.models.signals import post_migrate
```
Consider moving the imports to the top, unless there's a reason for it (circular import etc.)

sidmitra commented on tests/test_project/viewtest/tests.py:
```diff
@@ -438,3 +439,61 @@ def test_no_schema_list(self):
+
+    def tearDown(self):
+        """Restore original signal receivers and app config state"""
+        from django.db.models.signals import post_migrate
```
same here

mikicz commented:
@sidmitra do you want to review again?
